src/main/java/teammates/storage/api/NotificationsDb.java

```diff
@@ -37,6 +42,61 @@ public NotificationAttributes getNotification(String notificationId) {
         return notificationId.isEmpty() ? null : makeAttributesOrNull(getNotificationEntity(notificationId));
     }
 
+    /**
+     * Gets all notifications.
+     */
+    public List<NotificationAttributes> getAllNotifications() {
+        List<Notification> notifications = load().list();
+        List<NotificationAttributes> notificationAttributes = makeAttributes(notifications);
+        NotificationAttributes.sortByStartTime(notificationAttributes);
+        return notificationAttributes;
+    }
+
+    /**
+     * Gets notifications by {@code targetUser}.
+     *
+     * @return a list of notifications for the specified targetUser.
+     */
+    public List<NotificationAttributes> getNotificationsByTargetUser(NotificationTargetUser targetUser) {
+        assert targetUser != null;
+
+        List<Notification> endEntities = load()
+                .filter("targetUser", targetUser)
+                .filter("endTime >", Instant.now())
+                .list();
+
+        List<Notification> startEntities = load()
+                .filter("targetUser", targetUser)
+                .filter("startTime <", Instant.now())
+                .list();
+
+        if (targetUser != NotificationTargetUser.GENERAL) {
```

Comment:
Is this check redundant? I don't think `GENERAL` is an allowed value for `targetUser` parameter from request. It should only be student or instructor (checks should be done at the GET request handler).